src/txreconciliation.h

```diff
@@ -0,0 +1,585 @@
+    }
+
+    /**
+     * Reconciliation involves computing a space-efficient representation of transaction identifiers
+     * (a sketch). A sketch has a capacity meaning it allows reconciling at most a certain number
+     * of elements (see BIP-330). Considering whether we are going to send a sketch to a peer or use
+     * locally, we estimate the set difference.
+     */
+    Minisketch ComputeSketch(uint16_t capacity, bool use_snapshot=false)
+    {
+        Minisketch sketch;
+        std::set<uint256> working_set;
+        if (use_snapshot) {
+            working_set = m_local_set_snapshot;
+        } else {
+            working_set = m_local_set;
+            m_capacity_snapshot = capacity;
+        }
+        // Avoid serializing/sending an empty sketch.
+        if (working_set.size() == 0 || capacity == 0) return sketch;
+
+        std::vector<uint32_t> short_ids;
+        for (const auto& wtxid: working_set) {
+            uint32_t short_txid = ComputeShortID(wtxid);
+            short_ids.push_back(short_txid);
+            m_local_short_id_mapping.emplace(short_txid, wtxid);
+        }
+
+        capacity = std::min(capacity, MAX_SKETCH_CAPACITY);
+        sketch = Minisketch(RECON_FIELD_SIZE, 0, capacity);
+        if (sketch) {
+            for (const uint32_t short_id: short_ids) {
+                sketch.Add(short_id);
+            }
+        }
+        return sketch;
+    }
+
+public:
+
+    ReconciliationState(bool requestor, bool responder, bool flood_to, uint64_t k0, uint64_t k1) :
+        m_requestor(requestor), m_responder(responder), m_flood_to(flood_to),
+        m_k0(k0), m_k1(k1), m_local_q(DEFAULT_RECON_Q) {}
+
+    bool IsChosenForFlooding() const
+    {
+        return m_flood_to;
+    }
+
+    bool IsRequestor() const
+    {
+        return m_requestor;
+    }
+
+    bool IsResponder() const
+    {
+        return m_responder;
+    }
+
+    std::vector<uint256> GetLocalSet(bool snapshot) const
+    {
+        if (snapshot) {
+            return std::vector<uint256>(m_local_set_snapshot.begin(), m_local_set_snapshot.end());
+        } else {
+            return std::vector<uint256>(m_local_set.begin(), m_local_set.end());
+        }
+    }
+
+    uint16_t GetLocalSetSize(bool snapshot) const
+    {
+        if (snapshot) return m_local_set_snapshot.size();
+        return m_local_set.size();
+    }
+
+    uint16_t GetLocalQ() const
+    {
+        return m_local_q * Q_PRECISION;
+    }
+
+    std::vector<uint8_t> GetRemoteSketchSnapshot() const
+    {
+        return m_remote_sketch_snapshot;
+    }
+
+    uint16_t GetCapacitySnapshot() const
+    {
+        return m_capacity_snapshot;
+    }
+
+    ReconciliationPhase GetIncomingPhase() const
+    {
+        return m_incoming_recon;
+    }
+
+    ReconciliationPhase GetOutgoingPhase() const
+    {
+        return m_outgoing_recon;
+    }
+
+    std::chrono::microseconds GetNextRespond() const
+    {
+        return m_next_recon_respond;
+    }
```

Comment:
I think you can save a lot of those helpers by definitely making `TxReconciliationTracker` a friend of `ReconciliationState` ?